test_utilities/src/common.rs

```diff
@@ -5,16 +5,151 @@ use chrono::Utc;
 
 use qovery_engine::cloud_provider::utilities::sanitize_name;
 use qovery_engine::models::{
-    Action, Application, Context, Database, DatabaseKind, Environment, GitCredentials, Route, Router, Storage,
-    StorageType,
+    Action, Application, Context, Database, DatabaseKind, Environment, EnvironmentAction, GitCredentials, Route,
+    Router, Storage, StorageType,
 };
 
-use crate::utilities::{generate_id, generate_password, get_svc_name};
+use crate::cloudflare::dns_provider_cloudflare;
+use crate::utilities::{
+    generate_id, generate_password, get_environment_test_kubernetes, get_svc_name, FuncTestsSecrets,
+};
 use base64;
-use qovery_engine::cloud_provider::Kind;
+use qovery_engine::cloud_provider::aws::AWS;
+use qovery_engine::cloud_provider::digitalocean::DO;
+use qovery_engine::cloud_provider::models::NodeGroups;
+use qovery_engine::cloud_provider::scaleway::Scaleway;
+use qovery_engine::cloud_provider::{CloudProvider, Kind};
+use qovery_engine::engine::Engine;
 use qovery_engine::models::DatabaseMode::CONTAINER;
+use qovery_engine::transaction::{DeploymentOption, TransactionResult};
 use std::collections::BTreeMap;
 
+pub trait Cluster<T, U> {
+    fn docker_cr_engine(context: &Context) -> Engine;
+    fn cloud_provider(context: &Context) -> Box<T>;
+    fn kubernetes_nodes() -> Vec<NodeGroups>;
+    fn kubernetes_cluster_options(secrets: FuncTestsSecrets, cluster_id: Option<String>) -> U;
+}
+
+pub trait Infrastructure {
```

Comment:
This one rather sounds like a `Environment` trait isn't it?
If so, I would rename it to `Environment`

Reply:
At least I am not sure to grasp diff between Environment and Infrastructure here